src/java.base/share/classes/sun/security/ec/ECPrivateKeyImpl.java

```diff
@@ -92,48 +92,35 @@ public final class ECPrivateKeyImpl extends PKCS8Key implements ECPrivateKey {
             throws InvalidKeyException {
         this.s = s;
         this.params = params;
-        makeEncoding(s);
-
+        byte[] sArr = s.toByteArray();
+        // convert to fixed-length array
+        int numOctets = (params.getOrder().bitLength() + 7) / 8;
+        byte[] sOctets = new byte[numOctets];
+        int inPos = Math.max(sArr.length - sOctets.length, 0);
+        int outPos = Math.max(sOctets.length - sArr.length, 0);
+        int length = Math.min(sArr.length, sOctets.length);
+        System.arraycopy(sArr, inPos, sOctets, outPos, length);
+        Arrays.fill(sArr, (byte) 0);
+        makeEncoding(sOctets);
+        Arrays.fill(sOctets, (byte) 0);
     }
 
     ECPrivateKeyImpl(byte[] s, ECParameterSpec params)
             throws InvalidKeyException {
         this.arrayS = s.clone();
         this.params = params;
-        makeEncoding(s);
-    }
-
-    private void makeEncoding(byte[] s) throws InvalidKeyException {
-        algid = new AlgorithmId
-                (AlgorithmId.EC_oid, ECParameters.getAlgorithmParameters(params));
-        DerOutputStream out = new DerOutputStream();
-        out.putInteger(1); // version 1
-        byte[] privBytes = s.clone();
+        byte[] privBytes = arrayS.clone();
         ArrayUtil.reverse(privBytes);
-        out.putOctetString(privBytes);
+        makeEncoding(privBytes);
         Arrays.fill(privBytes, (byte) 0);
```

Comment:
You need a try-finally block so that the array is cleared even in the case an exception is thrown.